modules/dnn/src/ie_ngraph.cpp

```diff
@@ -504,12 +567,20 @@ void InfEngineNgraphNet::createNet(Target targetId) {
     }
 }
 
+#if INF_ENGINE_VER_MAJOR_LT(INF_ENGINE_RELEASE_2022_1)
+static inline
+InferenceEngine::Layout estimateLayout(size_t dims);
+#endif
+
 void InfEngineNgraphNet::init(Target targetId)
 {
     if (!hasNetOwner)
     {
         if (targetId == DNN_TARGET_OPENCL_FP16)
         {
+#if INF_ENGINE_VER_MAJOR_GE(INF_ENGINE_RELEASE_2022_1)
+            ov::pass::ConvertFP32ToFP16().run_on_model(ngraph_function);
```

Comment:
All FP16 tests are failed:
```
[ RUN      ] DNNTestNetwork.MobileNet_SSD_Caffe/2, where GetParam() = NGRAPH/OCL_FP16
[ INFO:0@17.101] global net_openvino.cpp:676 switchToOpenVINOBackend DNN: switching to OpenVINO backend... (networkID=151)
unknown file: Failure
C++ exception with description "Check 'false' failed at core/src/preprocess/pre_post_process.cpp:78:
Model doesn't have input with name data
" thrown in the test body.
[  FAILED  ] DNNTestNetwork.MobileNet_SSD_Caffe/2, where GetParam() = NGRAPH/OCL_FP16 (106 ms)
```

Reply:
Fixed that. Problem is not related to FP32->FP16 conversion but triggered by access to input nodes by name for precision change. Related: https://github.com/openvinotoolkit/openvino/issues/14725